### PR TITLE
Adds CI for macOS.

### PR DESCRIPTION
**Pull request by lexicalunit**

### CODE REVIEW

**Comment by mloskot** [comment]
This is great idea!

**Comment by lexicalunit** [comment]
@mloskot ok, got things to pass! Wanna code review this PR?

File: .travis.yml [hunk 1]
```diff
@@ -9,27 +9,73 @@ branches:
...
+               wuMpTTsKK1SCzTHzmaf8BqZ1qJ9hYFtuX/Q8YInL1c27g5FUfvAWO1SlBAMOLuk+OqrX3lYb/yzf+3O1Ndvn\
+               2JRMx5QaONUd7jTWs04VXZzvBpNfZOPU0ypKm6NEIvN/Wjak979iSLnW7O/HjpsOgdXEAzpIqb/xRaoDtrKP\
+               IkPwoDbi2kbrZcw+/ile3iZ2m1l9MP4s79WDn/v1FtFBv+C4GI7ikXPG6qQ8J6yWyrOcnt1HZ9u+gdJ9E8rs\
+               ZpMDSaR2rSB70rJMCGXzeG14dRUCzZsnumJ55+aY6rr/TDljOCOHQdHYofg0v6p82aojhtUo35EmedYbwkrI\
+               c7E2QzfGuKL2qpBjsXFnXrTRRuwtY+XQI/YF6nGECHSD/3JQTEZv14ICPh6IiWF2fMqLvsxEBIk5sZahyBCy\
+               Y1peQqb7CvxG4v1nNy3M9hS+k3L7Si541UpborI03sWgEUKWl9dBOxAwdYBaI6o4o+NvfpRbBY8rZLaEy+hK\
+               n+UuglQBeunMB54dvZxpS7zbPkWgBrnhIWMz89Rx6DT97OsHJdgDNcUOPv1b6xRAQqs+Zd3+nZ2vQIkvZJU5\
+               FwljLRWCfdg="
+
+ref:
```
Comment (mloskot) [inline .travis.yml]: This is really nice. I had no idea about `ref`.
Is this YAML feature or `.travis.yml`?

File: .travis.yml [hunk 2]
```diff
@@ -39,153 +85,67 @@ matrix:
...
-  - if [[ "$DB" == "vertica" ]]; then sudo /opt/vertica/sbin/install_vertica --failure-threshold HALT --accept-eula --dba-user-password dbadmin --license CE -s localhost; fi
-  - if [[ "$DB" == "vertica" ]]; then sudo -u dbadmin /opt/vertica/bin/adminTools -t create_db --database ci -s localhost -p dbadmin; fi
-  - if [[ "$DB" == "vertica" ]]; then sudo -u dbadmin openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /home/dbadmin/ci/v_ci_node0001_catalog/server.key -out /home/dbadmin/ci/v_ci_node0001_catalog/server.crt -batch; fi
-  - if [[ "$DB" == "vertica" ]]; then sudo -u dbadmin chmod 600 /home/dbadmin/ci/v_ci_node0001_catalog/server.key /home/dbadmin/ci/v_ci_node0001_catalog/server.crt; fi
-  - if [[ "$DB" == "vertica" ]]; then sudo -u dbadmin /opt/vertica/bin/adminTools -t stop_db --database ci -p dbadmin; fi
-  - if [[ "$DB" == "vertica" ]]; then sudo -u dbadmin sh -c 'echo "EnableSSL = 1" > /home/dbadmin/ci/v_ci_node0001_catalog/vertica.conf'; fi
-  - if [[ "$DB" == "vertica" ]]; then sudo -u dbadmin /opt/vertica/bin/adminTools -t start_db --database ci -p dbadmin; fi
+  - ${TRAVIS_BUILD_DIR}/scripts/ci/travis/before_install.$TRAVIS_OS_NAME.sh
```
Comment (mloskot) [inline .travis.yml]: I like the bash commands rationalisation with scripts

File: scripts/ci/travis/before_install.linux.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash -ue
+
+sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 20
+sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-5 20
+sudo update-alternatives --config gcc
+sudo update-alternatives --config g++
```
Comment (mloskot) [inline scripts/ci/travis/before_install.linux.sh]: It looks good. In future we may want to add `clang` builds on Linux too.

File: test/test_case_fixture.h [hunk 4]
```diff
@@ -374,10 +374,14 @@ struct test_case_fixture : public base_test_fixture
             REQUIRE(columns.column_name() == NANODBC_TEXT("c3"));
             if (vendor_ == database_vendor::sqlite)
             {
-#ifdef _WIN32
+                // FIXME: SQLite ODBC mis-reports decimal digits? Causing
+                //        columns.column_size() == 3.
+#if defined _WIN32
```
Comment (mloskot) [inline test/test_case_fixture.h]: Minor, style: `#if defined(_WIN32)` would be consistent with currently used preprocessor style.

File: test/test_case_fixture.h [hunk 5]
```diff
@@ -374,10 +374,14 @@ struct test_case_fixture : public base_test_fixture
...
                 REQUIRE(columns.sql_data_type() == -9); // FIXME: What is this type?
-                REQUIRE(
-                    columns.column_size() == 3); // FIXME: SQLite ODBC mis-reports decimal digits?
+                REQUIRE(columns.column_size() == 3);
+#elif defined __APPLE__
+                REQUIRE(columns.sql_data_type() == SQL_VARCHAR);
```
Comment (mloskot) [inline test/test_case_fixture.h]: Fun with ODBC drivers never ends :-)

File: .travis.yml (hunk 1 above)
Comment (lexicalunit) [inline .travis.yml]: I just made up the name `ref` and used it as a place to define a bunch of YAML references like:

```
  sources: &sources
    - boost-latest
    - george-edison55-precise-backports
    - ubuntu-toolchain-r-test
```

Which allows us to use `*sources` anywhere in the YAML file to inject the values from that reference.

Comment (mloskot) [inline .travis.yml]: Awesome trick!

**Pull request merged**